apps/docs/pages/guides/getting-started/tutorials/with-kotlin.mdx

```diff
@@ -0,0 +1,785 @@
+Read value from `BuildConfig` (Remember to import the right class, it is one from your project). For example:
+```kotlin
+val url = BuildConfig.SUPABASE_URL
+val apiKey = BuildConfig.API_KEY
+```
+
+### 3. Set up Supabase dependencies including Postgrest, GoTrue, Storage
+<img width="720" alt="Screen Shot 2023-06-05 at 21 39 24" src="https://github.com/hieuwu/product-sample-supabase-kt/assets/43868345/3f154fda-aa9e-4baa-81f2-5b744c7ad2fe"></img>
+
+In build.gradle (app) file, add these dependencies then press Sync now
+```kotlin
+// Supabase setup
+implementation("io.ktor:ktor-client-cio:1.6.0")
+implementation("io.github.jan-tennert.supabase:postgrest-kt-android:1.0.0")
+implementation("io.github.jan-tennert.supabase:storage-kt:1.0.0")
+implementation "io.ktor:ktor-client-android:2.2.1"
+implementation "io.ktor:ktor-utils-jvm:1.6.0"
+implementation "io.ktor:ktor-client-logging-jvm:1.6.0"
+implementation("io.ktor:ktor-client-core:2.2.4")
+```
+
+### 4. Set up Hilt for dependency injection
+In build.gradle (app) file, add these
+```kotlin
+implementation "com.google.dagger:hilt-android:$hilt_version"
+annotationProcessor "com.google.dagger:hilt-compiler:$hilt_version"
+implementation("androidx.hilt:hilt-navigation-compose:1.0.0")
+```
+Create new `ManageProductApplication.kt` class extending Application with `@HiltAndroidApp` annotation
+```kotlin
+// ManageProductApplication.kt
+@HiltAndroidApp
+class ManageProductApplication: Application()
+```
+Open AndroidManifest.xml file, update name property of Application tag
+```xml
+<application
+...
+    android:name=".ManageProductApplication"
+...
+</application>
+
+```
+Add annotation for MainActivity
+```kotlin
+@AndroidEntryPoint
+class MainActivity : ComponentActivity() {
+}
+```
+### 5. Provide Supabase instances with Hilt
+You can create it by just calling the constructor. However to make the app more testable, these should be provided
+Create SupabaseMdule.kt
+```kotlin
+@InstallIn(SingletonComponent::class)
+@Module
+object SupabaseModule {
+
+    @Provides
+    @Singleton
+    fun provideSupabaseClient(): SupabaseClient {
+        return createSupabaseClient(
+            supabaseUrl = BuildConfig.SUPABASE_URL,
+            supabaseKey = BuildConfig.API_KEY
+        ) {
+            install(Postgrest)
+            install(GoTrue)
+            install(Storage)
+        }
+    }
+
+    @Provides
+    @Singleton
+    fun provideSupabaseDatabase(client: SupabaseClient): Postgrest {
+        return client.postgrest
+    }
+    
+    @Provides
+    @Singleton
+    fun provideSupabaseGoTrue(client: SupabaseClient): GoTrue {
+        return client.gotrue
+    }
+
+
+    @Provides
+    @Singleton
+    fun provideSupabaseStorage(client: SupabaseClient): Storage {
+        return client.storage
+    }
+
+}
+```
+
+### 6. Create data transfer object (DTO) to get data
+Create `ProductDto.kt` class and use annotations to parse data from Supabase
+```kotlin
+@Serializable
+data class ProductDto(
```

Comment:
I suppose this part is not completely done, but in the preview, there is sql code to create a table for user profiles and not products. Also, is there a reason that `image` and `id` are nullable, but default to an empty string?

Reply:
I misused the sql query from other sample, I updated with mine. My bad for making the id nullable just updated.
My purpose of making `image` nullable is that at the beginning, we might not create a product with an image. After that, we can update that image.

Reply:
All good, it was just irritating that they both default to an empty string rather than null as they were nullable

Reply:
You are correct. It should be null instead of empty string. Updated for all places